File: ios/ReactNativeSplashView/ReactNativeSplashView.m
// ✅ Legacy (RCT_EXTERN) - ReactNativeSplashView.m
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(ReactNativeSplashView, NSObject)

RCT_EXTERN_METHOD(show)
RCT_EXTERN_METHOD(hide)
RCT_EXTERN_METHOD(isUsingLottie: (RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/ReactNativeSplashView/ReactNativeSplashView.swift
import UIKit

@objc(ReactNativeSplashView)
class ReactNativeSplashView: NSObject {

  @objc func hide() {
    Splash.hide()
  }

  @objc func isUsingLottie() -> Bool {
    return Splash.isUsingLottie()
  }
}
